src/qos_host/src/host.rs

```
@@ -0,0 +1,291 @@
+		)?;
+
+		let manifest_envelope = match manifest_envelope_resp {
+			ProtocolMsg::ManifestEnvelopeResponse { manifest_envelope } => {
+				*manifest_envelope
+			}
+			other => {
+				return Err(
+					Error(format!("unexpected response: expected a ProtocolMsg::ManifestEnvelopeResponse, but got: {other:?}"))
+				);
+			}
+		};
+
+		let vitals_log = if let Some(m) = manifest_envelope.as_ref() {
+			serde_json::to_string(&EnclaveVitalStats {
+				phase,
+				namespace: m.manifest.namespace.name.clone(),
+				nonce: m.manifest.namespace.nonce,
+				pivot_hash: m.manifest.pivot.hash,
+				pcr0: m.manifest.enclave.pcr0.clone(),
+				pivot_args: m.manifest.pivot.args.clone(),
+			})
+			.expect("always valid json. qed.")
+		} else {
+			serde_json::to_string(&phase).expect("always valid json. qed.")
+		};
+		println!("{vitals_log}");
+
+		let info = EnclaveInfo { phase, manifest_envelope };
+
+		Ok(Json(info))
+	}
+
+	/// Message route handler.
+	async fn message(
+		State(state): State<Arc<QosHostState>>,
+		encoded_request: Bytes,
+	) -> impl IntoResponse {
+		if encoded_request.len() > MAX_ENCODED_MSG_LEN {
+			return (
+				StatusCode::BAD_REQUEST,
+				borsh::to_vec(&ProtocolMsg::ProtocolErrorResponse(
+					ProtocolError::OversizeMsg,
+				))
+				.expect("ProtocolMsg can always serialize. qed."),
+			);
+		}
+
+		// DEBUG: remove later
+		match ProtocolMsg::try_from_slice(&encoded_request) {
+			Ok(r) => eprintln!("Received message: {r}"),
+			Err(e) => eprintln!("Unable to decode request: {e}"),
+		}
+
+		match state.enclave_client.call(&encoded_request).await {
+			Ok(encoded_response) => {
+				// DEBUG: remove later
+				match ProtocolMsg::try_from_slice(&encoded_response) {
+					Ok(r) => {
+						eprintln!("Enclave responded with: {r}");
+					}
+					Err(e) => {
+						eprintln!("Error deserializing response from enclave, make sure qos_host version match qos_core: {e}");
+					}
+				};
```

we could also return an error here? But generally think we should just remove